spec/jwt_spec.rb

```diff
@@ -332,7 +332,7 @@
 
     it 'raises "No verification key available" error' do
       token = JWT.encode({}, 'foo')
-      expect { JWT.decode(token, nil, true) }.to raise_error(JWT::DecodeError, 'No verification key available')
```

Comment:
Did the behaviour change in the test of why was the empty block added? We should be careful when changing old behaviour.

Reply:
Nice catch, have made some changes (moved none and no key verification from `Signature#verify` to `Decode#verify_signature`) to ensure there are no breaking behavioural changes.